hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -402,6 +403,25 @@ public Response put(
     }
   }
 
+  /**
+   * Validate S3 key path.
+   * disallows names containing ':'.
+   * Throws OS3Exception if invalid.
+   */
+  public static void isValidKeyPath(String path) throws OS3Exception {
+
+    // Split path components
+    String[] components = path.split("/");
+
+    for (String component : components) {
+      if (component.contains(":")) {
```

Comment:
Instead of just checking for a ":", we should do so a regex check like what's done in HDDS-5161.
Also add a unit test to show that an invalid key name gets created before your fix is not being allowed now.